src/IndustrialRobot/IndustrialRobot/ViewModels/MainViewModel.cs

```diff
@@ -1,14 +1,97 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Shapes;
+using IndustrialRobot.Views;
+using GalaSoft.MvvmLight.Command;
+using IndustrialRobot.Views.Interfaces;
 
 namespace IndustrialRobot.ViewModels
 {
-    public class MainViewModel : Screen
+    public class MainViewModel : Window 
     {
+        public SerialPort serialPort = new SerialPort();
+        public ICommand StartButtonCommand { get; set; }        
+        public ICommand SettingsButtonCommand { get; set; }
+        public ICommand ExitButtonCommand { get; set; }
 
+        //public RelayCommand<ICloseable> CloseWindowCommand { get; private set; }
+
+        public MainViewModel()
+        {
+            StartButtonCommand = new RelayCommand(ExecuteStartButton, CanExecuteStartButton);
+            SettingsButtonCommand = new RelayCommand(ExecuteSettingsButton, CanExecuteSettingsButton);
+            ExitButtonCommand = new RelayCommand(ExecuteExitButton, CanExecuteExitButton);
+            //CloseWindowCommand = new RelayCommand<ICloseable>(CloseWindow);     
+        }
+        private bool CanExecuteStartButton(object parameter)
+        {
+            return true;
+        }
+
+        private void ExecuteStartButton(object parameter)
+        {
+            serialPort.Close();
+            try
+            {
+                if (serialPort.IsOpen == false)
+                {
+                    serialPort.PortName = Properties.Settings.Default.Port;
+                    serialPort.BaudRate = Properties.Settings.Default.BaudRate;
+                    serialPort.Parity = Properties.Settings.Default.Parity;
+                    serialPort.DataBits = Properties.Settings.Default.DataBits;
+                    serialPort.StopBits = Properties.Settings.Default.StopBits;
+                    serialPort.Open();
+                }
+                ControlPanelView controlPanelView = new ControlPanelView(this);
+                serialPort.Write("SP 1" + "\r");
+                serialPort.Write("WH" + "\r");
+                controlPanelView.ShowDialog();
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(ex.Message);
+            }
+        }
+        private bool CanExecuteSettingsButton(object parameter)
+        {
+            return true;
+        }
+
+        private void ExecuteSettingsButton(object parameter)
+        {
+            SettingsView settingsWindow = new SettingsView();
+            settingsWindow.Show();
+        }
+        private bool CanExecuteExitButton(object parameter)
+        {
+            return true;
+        }
+
+        private void ExecuteExitButton(object parameter)
+        {
+            serialPort.Close();
+            Close(); //nie zamyka okna!
```

Comment:
czyzbys po to potrzebowal dziedziczenie z Window? Rzuc okiem np tu, spoko to ziutek opisal: https://web.csulb.edu/~pnguyen/cecs475/pdf/closingwindowmvvm.pdf

Reply:
Dokładnie tak miałem wcześniej zrobione, jest zakomentowane nawet w tym kodzie. Tylko jak tak robiłem to aplikacja w dziwny sposób się zachowywała. To jest po kliknięciu tego przycisku zamykało się okno, ale de facto aplikacja dalej działała.